pkg/controller/common/helpers.go

```diff
@@ -285,12 +298,41 @@ func ValidateIgnition(ignconfig interface{}) error {
 	}
 }
 
+// InSlice search for an element in slice and return true if found, otherwise return false
+func InSlice(elem string, slice []string) bool {
+	for _, k := range slice {
+		if k == elem {
+			return true
+		}
+	}
+	return false
+}
+
+func validateExtensions(exts []string) error {
+	supportedExtensions := []string{"usbguard", "kernel-devel"}
```

Comment:
Could these be constants exported alongside the MC api types? If we could even add them to the CRD's openapi schema that'd be even better, so that users get immediate feedback on invalid extensions.

Reply:
Not very sure. The list of supported extensions is going to change whenever we add more extensions support in upcoming releases. In future, we are also exploring to add metadata for supported extension list in machine-os-content https://github.com/openshift/os/issues/409 which MCO can read from and validate before rendering.